wdl-lint/src/rules/redundant_input_assignment.rs

```diff
@@ -0,0 +1,90 @@
+//! A lint rule for flagging redundant input assignments
+
+use std::fmt::Debug;
+use wdl_ast::{AstToken, VersionStatement};
+use wdl_ast::Diagnostic;
+use wdl_ast::Diagnostics;
+use wdl_ast::Document;
+use wdl_ast::Span;
+use wdl_ast::SupportedVersion;
+use wdl_ast::v1::{Decl, InputSection};
+use wdl_ast::VisitReason;
+use wdl_ast::Visitor;
+
+use crate::Rule;
+use crate::Tag;
+use crate::TagSet;
+
+/// The identifier for the Redundant Input Assignment rule.
+const ID: &str = "RedundantInputAssignment";
+
+/// Create a "Redundant Input Assignment" diagnostic.
+fn redundant_input_assignment(span: Span) -> Diagnostic {
+    Diagnostic::note("redundant input assignments can be shortened")
+        .with_rule(ID)
+        .with_label("redundant input assignments can be shortened", span)
+}
+
+/// Detects a malformed lint directive.
+#[derive(Default, Debug, Clone, Copy)]
+pub struct RedundantInputAssignment {
+    included_version: bool,
+}
+
+impl Rule for RedundantInputAssignment {
+    fn id(&self) -> &'static str {
+        ID
+    }
+
+    fn description(&self) -> &'static str {
+        "Flags redundant input assignments."
+    }
+
+    fn explanation(&self) -> &'static str {
+        "Input assignments that are redundant can be shortened. For example, `{ input: a = a }` \
+        can be shortened to `{ input: a }`."
+    }
+
+    fn tags(&self) -> TagSet {
+        TagSet::new(&[Tag::Clarity, Tag::Correctness])
+    }
+
+    fn exceptable_nodes(&self) -> Option<&'static [wdl_ast::SyntaxKind]> {
+        None
+    }
+}
+
+impl Visitor for RedundantInputAssignment {
+    type State = Diagnostics;
+
+    fn document(&mut self, _: &mut Self::State, _: VisitReason, _: &Document, _: SupportedVersion) {
+        // This is intentionally empty, as this rule has no state.
+    }
+
+    fn version_statement(&mut self, state: &mut Self::State, reason: VisitReason, stmt: &VersionStatement) {
+        // this current takes into account for any possible patch versions, which might be unnecessary
+        let binding = stmt.version();
+        let split_versions = binding.as_str().split(".").collect::<Vec<_>>();
+        if split_versions[0] == "1" && split_versions[1] == "0" {
+            self.included_version = false;
+        } else {
+            self.included_version = true;
+        }
+    }
```

Comment:
@a-frantz I wrote this function to account for patch versions, but if it isn't necessary, I can simply do a check for == "1.0"? I'm not the most familiar with current WDL SemVer

Reply:
WDL version statements do not include a patch number (but there are patch releases, which can be confusing. The idea is that patch releases are mostly fixing typos and making clarifications in the text, which don't impact parsing or semantics, so the patch can be ignored in this context). A WDL version statement will always look like `version 1.2` or `version 1.0`, where the version is just `<major>.<minor>`.

Reply:
Are there any versions to consider prior to "1.0"? If there aren't zero versions, I think this function can be less verbose

Reply:
There are "draft" releases before 1.0, but we as a team have explicitly decided we will not support them. So you can assume any version will be `>=v1.0`. If someone tries to run our tool on a draft version, they will be hit with an error long before reaching your part of the code.